crates/corro-types/src/pubsub.rs

```diff
@@ -166,153 +187,14 @@ impl SubsManager {
-                        });
-                    }
-                    mpsc::error::TrySendError::Closed(_) => {
-                        if let Some(handle) = self.remove(id) {
-                            tokio::spawn(handle.cleanup());
-                        }
-                    }
-                }
-            }
-        }
-
-        Ok(())
-    }
-
-    pub fn match_changes(&self, changes: &[Change], db_version: CrsqlDbVersion) {
-        trace!(
-            %db_version,
-            "trying to match changes to subscribers, len: {}",
-            changes.len()
-        );
-        if changes.is_empty() {
-            return;
-        }
-        let handles = {
-            let inner = self.0.read();
-            if inner.handles.is_empty() {
-                return;
-            }
-            inner.handles.clone()
-        };
-
-        for (id, handle) in handles.iter() {
-            trace!(sub_id = %id, %db_version, "attempting to match changes to a subscription");
-            let mut candidates = MatchCandidates::new();
-            let mut match_count = 0;
-            for change in changes.iter().map(MatchableChange::from) {
-                if handle.filter_matchable_change(&mut candidates, change) {
-                    match_count += 1;
-                }
-            }
-
-            // metrics...
-            for (table, pks) in candidates.iter() {
-                counter!("corro.subs.changes.matched.count", "sql_hash" => handle.inner.hash.clone(), "table" => table.to_string()).increment(pks.len() as u64);
-            }
-
-            trace!(sub_id = %id, %db_version, "found {match_count} candidates");
-
-            if let Err(e) = handle.inner.changes_tx.try_send((candidates, db_version)) {
-                error!(sub_id = %id, "could not send change candidates to subscription handler: {e}");
-                match e {
-                    mpsc::error::TrySendError::Full(item) => {
-                        warn!("channel is full, falling back to async send");
-
-                        let changes_tx = handle.inner.changes_tx.clone();
-                        tokio::spawn(async move {
-                            _ = changes_tx.send(item).await;
-                        });
-                        counter!("corro.subs.changes.channel.async_fallbacks_count", "sql_hash" => handle.inner.hash.clone()).increment(1);
-                    }
-                    mpsc::error::TrySendError::Closed(_) => {
-                        if let Some(handle) = self.remove(id) {
-                            tokio::spawn(handle.cleanup());
-                        }
-                    }
-                }
-            }
-        }
-    }
 }
 
 #[derive(Debug)]
-struct MatchableChange<'a> {
-    table: &'a TableName,
-    pk: &'a [u8],
-    column: &'a ColumnName,
+pub struct MatchableChange<'a> {
+    pub table: &'a TableName,
+    pub pk: &'a [u8],
+    pub column: &'a ColumnName,
+    pub cl: &'a i64,
```

Comment:
I think using a `i64` directly instead of a reference is best here. I believe they have the same size and it's just easier to deal with integers directly than references to them imo.